tests/src/test/scala/whisk/core/controller/test/WebActionsApiTests.scala

```diff
@@ -126,36 +122,47 @@ trait WebActionsApiBaseTests extends ControllerTestCommon with BeforeAndAfterEac
   val systemKey = BasicAuthenticationAuthKey(uuid, Secret())
   val systemIdentity =
     Future.successful(Identity(systemId, Namespace(EntityName(systemId.asString), uuid), systemKey, Privilege.ALL))
+  val namespace = EntityPath(systemId.asString)
+  val proxyNamespace = namespace.addPath(EntityName("proxy"))
   override lazy val entitlementProvider = new TestingEntitlementProvider(whiskConfig, loadBalancer)
   protected val testRoutePath = webInvokePathSegments.mkString("/", "/", "")
+  def aname() = MakeName.next("web_action_tests")
 
   behavior of "Web actions API"
 
-  var failActionLookup = false // toggle to cause action lookup to fail
   var failActivation = 0 // toggle to cause action to fail
   var failThrottleForSubject: Option[Subject] = None // toggle to cause throttle to fail for subject
+  var failCheckEntitlement = false // toggle to cause entitlement to fail
   var actionResult: Option[JsObject] = None
-  var requireAuthentication = false // toggle require-whisk-auth annotation on action
   var requireAuthenticationAsBoolean = true // toggle value set in require-whisk-auth annotation (true or  requireAuthenticationKey)
+  var testParametersInInvokeAction = true // toggle to test parameter in invokeAction
   var requireAuthenticationKey = "example-web-action-api-key"
-  var customOptions = true // toogle web-custom-options annotation on action
   var invocationCount = 0
   var invocationsAllowed = 0
 
+  override def beforeAll() = {
+    implicit val tid = transid()
+
+    put(entityStore, stubPackage, garbageCollect = false)
```

Comment:
this is why the stubs arent collected: garbageCollect is set to false.

Reply:
yes, it seems other test case remove all document after tests.

https://github.com/apache/incubator-openwhisk/blob/master/tests/src/test/scala/whisk/core/controller/test/ControllerTestCommon.scala#L203